README.md

```diff
@@ -64,15 +76,14 @@ test.cb('that chat app can be mocked', t => {
     });
   });
 
-  const app = new ChatApp(fakeURL);
+  const app = await ChatApp.connect(fakeURL);
   app.sendMessage('test message from app'); // NOTE: this line creates a micro task
 
-  // NOTE: this timeout is for creating another micro task that will happen after the above one
-  setTimeout(() => {
-    t.is(app.messages.length, 1);
-    t.is(app.messages[0], 'test message from mock server', 'we have stubbed our websocket backend');
-    mockServer.stop(t.done);
-  }, 100);
+  await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
It's better to avoid using `setTimeout` at all. I think it's worth adding some helpers.
```
const waitFor = (connection, event) => new Promise(resolve => {
    const handler = () => {
        resolve();
        connection.removeEventListener(event, handler);
    }
    connection.addEventListener(event, handler);
});


const app = new ChatApp(fakeURL);
await waitFor(app.connection, 'open');
...
await waitFor(app.connection, 'message');
```

Reply:
The are quite some issues with suggestion provided, so I used an alternative approach.